frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
So it won't read the whole document? Only no more than the first 10KB? Ok, in that case the overhead is negligible and it is fine to do on the first open of each document.